tests/benchmarking/test_run_benchmark.py

```diff
@@ -54,5 +58,142 @@ def test_run_benchmark_list_options(tmpdir):
+    failing_name = "jennrich_sampson"
+    failing = {failing_name: all_problems[failing_name]}
+
+    optimize_options = ["scipy_lbfgsb"]
+
+    with pytest.warns():
+        result = run_benchmark(problems=failing, optimize_options=optimize_options)
+
+    key = (failing_name, "scipy_lbfgsb")
+    assert isinstance(result[key]["solution"], str)
+
+
+# ======================================================================================
+# Custom benchmark functions
+# ======================================================================================
+
+
+def get_benchmark_problems_custom(raw_problems, internal_criterion):
+    problems = {}
+    for name, specification in raw_problems.items():
+        inputs = _create_problem_inputs_custom(specification, internal_criterion)
+
+        problems[name] = {
+            "inputs": inputs,
+            "solution": _create_problem_solution_custom(specification),
+            "info": specification.get("info", {}),
+        }
+
+    return problems
+
+
+def _create_problem_inputs_custom(specification, internal_criterion_func):
+    _criterion = partial(
+        internal_criterion_func,
+        criterion=specification["criterion"],
+    )
+
+    _params = specification["start_x"]
+    inputs = {"criterion": _criterion, "params": _params}
+
+    return inputs
+
+
+def _create_problem_solution_custom(specification):
+    _solution_x = specification.get("solution_x")
+    if _solution_x is None:
+        _solution_x = specification["start_x"] * np.nan
+
+    _value = specification["solution_criterion"]
+
+    solution = {
+        "params": _solution_x,
+        "value": _value,
+    }
+    return solution
+
+
+def _internal_criterion_pandas(params, criterion):
+    x = params["value"].to_numpy()
+    critval = criterion(x)
+
+    out = {
+        "root_contributions": critval,
+        "value": critval @ critval,
+    }
+
+    return out
+
+
+def _internal_criterion_dict(params, criterion):
+    x = params["value"]
+    critval = criterion(x)
+
+    out = {
+        "root_contributions": critval,
+        "value": critval @ critval,
+    }
+
+    return out
 
-    assert set(res) == expected_keys
+
+problems_pandas_input = {
+    "linear_full_rank_good_start": {
+        "criterion": partial(linear_full_rank, dim_out=45),
+        "start_x": pd.DataFrame(np.ones(9), columns=["value"]),
+        "solution_x": pd.DataFrame(linear_full_rank_solution_x, columns=["value"]),
+        "start_criterion": 72,
+        "solution_criterion": 36,
+    },
+    "rosenbrock_good_start": {
+        "criterion": rosenbrock,
+        "start_x": pd.DataFrame([-1.2, 1], columns=["value"]),
+        "solution_x": pd.DataFrame(np.ones(2), columns=["value"]),
+        "start_criterion": 24.2,
+        "solution_criterion": 0,
+    },
+}
+
+
+prolems_dict_input = {
+    "linear_full_rank_good_start": {
+        "criterion": partial(linear_full_rank, dim_out=45),
+        "start_x": {"value": np.ones(9)},
```

Comment:
This is not a very good dict test case because it behaves exactly like a typical DataFrame with "value" column. When we talk about dict params we mean something like: `{"a": 1, "b":  np.ones((2, 2, 2))}`. The name `"value"` is nothing special in the dict case!

Reply:
Done.